Depending on installing pybind11 using pip3 instead of apt

essamROCm opened the pull request:
Resolving issue[#119](https://github.com/ROCm/rocPyDecode/issues/119), able to find pybind11 package in any environment via CMake, depending on it being installed with 'pip3' instead of installing with 'apt' pybind11-dev.

The pybind11-dev apt installation is removed, and rocPyDecode build/runtime is tested on bare-metal and inside plain docker image (with ROCm).

Commit:
Depending on installing pybind11 using pip3 instead of 'apt pybind11-dev'

LakshmiKumar23 commented on CMakeLists.txt:
```diff
@@ -102,6 +102,21 @@ find_package(rocDecode QUIET)
 
+# get the pybind installation folder and pass it to the find package
+execute_process(
+    COMMAND sudo find / -name "pybind11Config.cmake" -not -path "*docker*"
```
@essamROCm please add a Findpybind11.cmake file in the [cmake directory](https://github.com/ROCm/rocPyDecode/tree/develop/cmake) instead of this. Refer to the other files to see how to find it

essamROCm replied:
Done in commit: https://github.com/ROCm/rocPyDecode/pull/122/commits/6e8904d3ada90ae034132f2c27a4b883b8a9d3d1

LakshmiKumar23 commented on rocPyDecode-requirements.py:
```diff
@@ -139,7 +139,7 @@ def ERROR_CHECK(call):
     'rocdecode-test',
     'python3-dev',
-    'pybind11-dev',
+    #'pybind11-dev', # use 'pip3 install pybind11' instead
```
We should remove this commented line and add the pip install as a step in the script

essamROCm replied:
Done in commit: https://github.com/ROCm/rocPyDecode/pull/122/commits/6e8904d3ada90ae034132f2c27a4b883b8a9d3d1

Commit:
Removed commented line in rocPyDecode-requirements.py
Added pip install pybind11 in rocPyDecode-requirements.py
Added cmake/Findpybind11.cmake file to find the pybind11
Removed find PATH from CMakeLists.txt

Commit:
Grantee finding one pybind11 package installation

Commit:
Avoid root level in finding python

LakshmiKumar23 commented on cmake/Findpybind11.cmake:
```diff
@@ -0,0 +1,59 @@
+# ################################################################################
+
+execute_process(
+    COMMAND sudo find / -name "pybind11Config.cmake" -not -path "*docker*"
```
We should not use commands in this script. Please refer to the other files. There is a method to find path. This should be redone

essamROCm replied:
find_path does not work for this purpose, whatever I tried with find_path() failed to locate the right path. 'execute_process' however works well and has no issue so far.

If you know a way to make 'find_path' work, please help with that, I appreciate your help.

essamROCm replied:
Done, changed to find_path: https://github.com/ROCm/rocPyDecode/pull/122/commits/04f3fafa23a53c267cefca2cf77f44b5e9d6d524

rrawther reviewed (commented):
Can we go over this PR in our daily syncup meeting?

Commit:
Using find_path instead of execute_process

essamROCm commented:
> Can we go over this PR in our daily syncup meeting?

Hi @rrawther 
Per our discussion in the meeting, '_find_path_' is used now successfully, with no issue.

essamROCm committed:
Merge branch 'develop' into ea/Using_pip_to_install_pybind11_not_apt

essamROCm committed:
Merge branch 'develop' into ea/Using_pip_to_install_pybind11_not_apt

essamROCm committed:
Merge branch 'develop' into ea/Using_pip_to_install_pybind11_not_apt

LakshmiKumar23 commented on cmake/Findpybind11.cmake:
```diff
@@ -0,0 +1,62 @@
+        FOUND_VAR pybind11_FOUND
+        REQUIRED_VARS pybind11_INCLUDE_DIR
+)
+message(STATUS "pybind11 was FOUND: ${pybind11_FOUND} --- HEADERS: ${pybind11_INCLUDE_DIR}"  )
```
please add a check, similar to this: https://github.com/ROCm/rocDecode/blob/81e341b786a939af9a0b8f13831640aa5fe0714a/cmake/FindLibva.cmake#L51
```
if (pybind11_FOUND)
   message("-- ${White}Using pybind11 \n\tIncludes: {pybind11_INCLUDE_DIR}" )
   include("${PYBIND11_CMAKE_FOLDER}/pybind11Targets.cmake")
   add_library(pybind11::headers IMPORTED INTERFACE)
   set_target_properties(pybind11::headers PROPERTIES INTERFACE_LINK_LIBRARIES pybind11::pybind11_headers)
   include("${PYBIND11_CMAKE_FOLDER}/pybind11Common.cmake")
else()
   message(FATAL_ERROR "{Red}Findpybind11 -- pybind11 NOT FOUND${ColourReset}")
endif()
```

essamROCm replied:
Done in this commit: https://github.com/ROCm/rocPyDecode/pull/122/commits/38943a6f59559e4b14ea817e3bccbd13767d7877

LakshmiKumar23 commented:
@essamROCm please change the README as well to reflect this

Commit:
Adding check for if pybind11_FOUND, and refelct on README

essamROCm commented:
> @essamROCm please change the README as well to reflect this

Done in this commit: https://github.com/ROCm/rocPyDecode/pull/122/commits/38943a6f59559e4b14ea817e3bccbd13767d7877

LakshmiKumar23 commented:
Please look at the CI failure: https://dev.azure.com/ROCm-CI/ROCm-CI/_build/results?buildId=11188&view=logs&j=7f9e356b-b957-5776-bc7e-49652acc6cbd&t=cd5d8c7d-ab0a-594b-babe-b96a7b8650ea&l=57

essamROCm commented:
> Please look at the CI failure: https://dev.azure.com/ROCm-CI/ROCm-CI/_build/results?buildId=11188&view=logs&j=7f9e356b-b957-5776-bc7e-49652acc6cbd&t=cd5d8c7d-ab0a-594b-babe-b96a7b8650ea&l=57

I talked to Joseph, he needs to make sure pybind11 is installed, he is adding it to his scripts now.

essamROCm commented:
> > Please look at the CI failure: https://dev.azure.com/ROCm-CI/ROCm-CI/_build/results?buildId=11188&view=logs&j=7f9e356b-b957-5776-bc7e-49652acc6cbd&t=cd5d8c7d-ab0a-594b-babe-b96a7b8650ea&l=57
> 
> I talked to Joseph, he needs to make sure pybind11 is installed, he is adding changes to his scripts.

https://github.com/ROCm/ROCm/pull/3954

@amd-jmacaran, Hi Joseph, would you please check again the CI-azure build failure.

essamROCm committed:
Merge branch 'develop' into ea/Using_pip_to_install_pybind11_not_apt

jayhawk-commits commented:
> > > Please look at the CI failure: https://dev.azure.com/ROCm-CI/ROCm-CI/_build/results?buildId=11188&view=logs&j=7f9e356b-b957-5776-bc7e-49652acc6cbd&t=cd5d8c7d-ab0a-594b-babe-b96a7b8650ea&l=57
> > 
> > 
> > I talked to Joseph, he needs to make sure pybind11 is installed, he is adding changes to his scripts.
> 
> [ROCm/ROCm#3954](https://github.com/ROCm/ROCm/pull/3954)
> 
> @amd-jmacaran, Hi Joseph, would you please check again the CI-azure build failure.

https://github.com/ROCm/ROCm/pull/3978 should resolve the issues. You can see the passing run in the checks. The old failing checks won't be replaced until PR 3978 is merged.

essamROCm committed:
Merge branch 'develop' into ea/Using_pip_to_install_pybind11_not_apt

essamROCm committed:
Merge branch 'develop' into ea/Using_pip_to_install_pybind11_not_apt

kiritigowda committed:
Merge branch 'develop' into ea/Using_pip_to_install_pybind11_not_apt
